src/ConflictResolver.cc

```diff
@@ -171,14 +176,16 @@ namespace cbl_internal {
 
         // Determine the resolution type:
         CBLDocument::Resolution resolution;
-        if (resolved == myDoc) {
+        if (resolved == localDoc)
             resolution = CBLDocument::Resolution::useLocal;
-            resolved = nullptr;
-        } else if (resolved == conflict) {
+        else if (resolved == conflict)
             resolution = CBLDocument::Resolution::useRemote;
-            resolved = nullptr;
-        } else {
+        else {
             if (resolved) {
+                // Need to explicitly release here because the resolved doc (merged doc) is created
+                // by the custom conflict resolver.
+                CBLDocument_Release(resolved);
```

Comment:
Don't release it until you are done using it.

Reply:
The retained count is actually 2 as it is bumped when it's assigned to RetainedConst<CBLDocument>. Pretty confusing.

Reply:
I will release it at the end which will make the code more understandable even though it's the same.